tools/export/gnuarmeclipse/__init__.py

```diff
@@ -244,7 +244,7 @@ def create_jinja_ctx(self):
             opts['ld']['system_libraries'] = self.system_libraries
             opts['ld']['script'] = join(id.capitalize(),
                                         "linker-script-%s.ld" % id)
-            opts['cpp_cmd'] = " ".join(toolchain.preproc)
+            opts['cpp_cmd'] = "".join('"{}"'.format(toolchain.preproc[0])) + " " + " ".join(toolchain.preproc[1:])
```

Comment:
Please remove the `join` call to something that is not a list. Please limit lines to 80 characters.

This should be:
```python 
opts['cpp_cmd'] = '"{}" {}'.format(toolchain.preproc[0], 
                                   " ".join(toolchain.preproc[1:]))
```

Reply:
Upon further thinking, I think we should just use the filename. So:
```python
opts['cpp_cmd'] = '{} {}'.format(basename(toolchain.preproc[0]), 
                                   " ".join(toolchain.preproc[1:]))
```
or
```python
opts['cpp_cmd'] = " ".join(basename(toolchain.preproc[0]) +
                           toolchain.preproc[1:])

Reply:
Okay i have removed the first join that was not okay. But sorry your changes are not correct. 
1) basename  is the filename only so that did not solve the issue at all.
2) you removed the  '"{}"'.form... this is the important part of the fix. ...

So I have updated my PR I do not see any easier Solution. Also I do not understand why 80 chars only because have you looked on line 429 as an example the line has 111 chars :-)

cheers

Reply:
> basename is the filename only so that did not solve the issue at all.

That's the point. When we run this exporter on the website, you have no path information to use, so we have to make an assumption. I'm suggesting that we unify these behaviors.

>you removed the '"{}"'.form... this is the important part of the fix. ...

Not if you invoke the command with not preceding path. There should be no spaces in the path if it's only `arm-none-eabi-cpp`.

Reply:
> Also I do not understand why 80 chars only because have you looked on line 429 as an example the line has 111 chars :-)

I'm trying not to add formatting work for later.